### DISCUSSION TITLE
Problem with view model instance preservation when navigating back using navigator extension

### DISCUSSION THREAD

**Discussion by LITTOMA** [unoplatform/uno.extensions#2034] [Q&A] [answered]
I'm working on an application using Uno Platform and facing a problem with navigation between ViewModels. My application includes two ViewModels, named ViewModelA and ViewModelB. I use the INavigator service for navigation from ViewModelA to ViewModelB. However, when I navigate back to ViewModelA using the `NavigateBackAsync` method, the application doesn't return to the original ViewModelA instance. Instead, it creates a new instance of ViewModelA. I'm seeking a solution that ensures returning to the original instance of ViewModelA when navigating back.

**Example Code:**
```csharp
// Navigation from ViewModelA to ViewModelB
public class ViewModelA : ViewModelBase
{
    private readonly INavigator _navigator;

    public ViewModelA(INavigator navigator)
    {
        _navigator = navigator;
    }

    public void NavigateToViewModelB()
    {
        _navigator.NavigateRouteAsync(this, "ViewModelB");
    }
}

// Navigation back to ViewModelA
public class ViewModelB : ViewModelBase
{
    private readonly INavigator _navigator;

    public ViewModelB(INavigator navigator)
    {
        _navigator = navigator;
    }

    public void NavigateBack()
    {
        _navigator.NavigateBackAsync(this);
    }
}
```

In this scenario, I use `NavigateRouteAsync` for moving from ViewModelA to ViewModelB and `NavigateBackAsync` for returning. The challenge is that it results in a new instance of ViewModelA rather than using the original one. What approach should I take to ensure that the original ViewModelA is reused upon navigating back?

**Comment by nickrandolph** [comment, marked as answer]
Hi @LITTOMA I think you need to set NavigationCacheMode on PageA to required. What's happening is the page is getting recreated (instead of a cached version being used), so you get a new instance of the viewmodel. 
Let me know if this fixes your issue

**Reply by LITTOMA** [reply]
Yes, this fixed mu issue. Thanks!
